src/api/BundlePart.ts

```diff
@@ -33,6 +33,26 @@ export interface IBundlePartDataType {
  */
 export class BundlePart {
 
+  /**
+   * Function for mangaling a name into something valid for CICS
+   *
+   * @throws ImperativeError
+   * @memberof BundlePart
+   */
+  protected static mangleName(text: string, maxLength: number): string {
+
+    // replace underscore with hyphen
+    const text2 = text.replace(/_/g, "-");
+
+    // Convert all unsupported characters to X
+    const text3 = text2.replace(/[^0-9,^A-Z,^a-z\-]/g, "X");
```

Comment:
I think this would be clearer if you used a single variable throughout - perhaps call it mangledName?

Reply:
Fair enough; that routine has passed through several different incarnations; this most recent change was just to make it static (and to move it earlier in the file so that the build wouldn't complain)). There were points in the past when the intermediate values were more interesting.